src/bootstrap/src/lib.rs

```diff
@@ -700,15 +700,13 @@ impl Build {
 
     /// Gets the space-separated set of activated features for the compiler.
     fn rustc_features(&self, kind: Kind, target: TargetSelection, crates: &[String]) -> String {
-        let _possible_features_by_crates: HashSet<_> = crates
+        let possible_features_by_crates: HashSet<_> = crates
             .iter()
             .flat_map(|krate| &self.crates[krate].features)
             .map(std::ops::Deref::deref)
             .collect();
-        let check = |_feature: &str| -> bool {
-            //crates.is_empty() || possible_features_by_crates.contains(feature)
-            // change necessary while resolver is broken
-            true
+        let check = |feature: &str| -> bool {
+            crates.is_empty() || possible_features_by_crates.contains(feature)
```

Comment:
`possible_features_by_crates` is constructed from `crates`, so iiuc `possible_features_by_crates` will be empty if `crates` is empty, correct?

Also `crates` is immutable.

If above is true, then it would be enough to only do `possible_features_by_crates.contains(feature)`, because if `possible_features_by_crates` is empty it will not contain `feature`.

Reply:
This change brings this code back in sync with upstream.

Also those do not mean the same thing, empty `possible_features_by_crates` means no crates have any features that are allowed, so the closure should always return false in that case, where as an empty `crates` means we will build all crates so any feature is fine to be passed.

> Also crates is immutable.

?